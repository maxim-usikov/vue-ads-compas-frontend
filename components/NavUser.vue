<template>
  <VList v-if="isAuthenticated" class="pa-1 text-truncate">
    <VListTile avatar>
      <VListTileAvatar
        color="blue-grey darken-4"
        class="text-uppercase white--text"
      >
        {{ me.email[0].toUpperCase() }}
      </VListTileAvatar>

      <VListTileContent>
        <VListTileTitle>{{ me.email }}</VListTileTitle>
      </VListTileContent>

      <VSpacer />

      <VListTileAction>
        <VMenu bottom left>
          <VBtn slot="activator" icon> <VIcon>more_vert</VIcon> </VBtn>
          <VList>
            <VListTile @click="handleSignOut">
              <VListTileTitle>Sign out</VListTileTitle>
            </VListTile>
          </VList>
        </VMenu>
      </VListTileAction>
    </VListTile>

    <VDivider />
  </VList>

  <VList v-else class="pt-0" dense>
    <VListTile @click="handleSignIn">
      <VListTileAction>
        <VIcon>person</VIcon>
      </VListTileAction>

      <VListTileContent>
        <VListTileTitle>Sign in</VListTileTitle>
      </VListTileContent>
    </VListTile>

    <VDivider />
  </VList>
</template>

<script>
import { mapGetters } from 'vuex'

export default {
  name: 'NavUser',

  computed: {
    ...mapGetters(['isAuthenticated', 'me'])
  },

  methods: {
    handleSignIn() {
      this.$router.push({ name: 'sign-in' })
    },

    handleSignOut() {
      this.$auth.logout()
      this.$router.push({ name: 'index' })
    }
  }
}
</script>
