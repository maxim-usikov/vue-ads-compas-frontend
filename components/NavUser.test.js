import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex, { Store } from 'vuex'

import NavUser from './NavUser.vue'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('NavUser', () => {
  let $c
  let getters
  let store

  beforeEach(() => {
    getters = {
      isAuthenticated: () => true,
      me: () => ({ email: 'test@test.test' })
    }

    store = new Store({
      getters
    })

    $c = shallowMount(NavUser, { store, localVue })
  })

  it('is a Vue instance', () => {
    expect($c.isVueInstance()).toBeTruthy()
  })

  it('name is `NavUser`', () => {
    expect($c.name()).toBe('NavUser')
  })

  it('renders correctly', () => {
    expect($c.element).toMatchSnapshot()
  })
})
