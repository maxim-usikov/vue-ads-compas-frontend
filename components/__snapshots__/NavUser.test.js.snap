// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavUser renders correctly 1`] = `
<vlist-stub
  class="pa-1 text-truncate"
>
  <vlisttile-stub
    activeclass="primary--text"
    avatar="true"
  >
    <vlisttileavatar-stub
      class="text-uppercase white--text"
      color="blue-grey darken-4"
      size="40"
    >
      
      T
    
    </vlisttileavatar-stub>
     
    <vlisttilecontent-stub>
      <vlisttiletitle-stub>
        test@test.test
      </vlisttiletitle-stub>
    </vlisttilecontent-stub>
     
    <vspacer-stub />
     
    <vlisttileaction-stub>
      <vmenu-stub
        bottom="true"
        closedelay="0"
        closeonclick="true"
        closeoncontentclick="true"
        contentclass=""
        left="true"
        maxheight="auto"
        maxwidth="auto"
        nudgebottom="0"
        nudgeleft="0"
        nudgeright="0"
        nudgetop="0"
        nudgewidth="0"
        opendelay="0"
        openonclick="true"
        origin="top left"
        transition="v-menu-transition"
      >
        <vbtn-stub
          activeclass="v-btn--active"
          icon="true"
          tag="button"
          type="button"
        >
          <vicon-stub>
            more_vert
          </vicon-stub>
        </vbtn-stub>
         
        <vlist-stub>
          <vlisttile-stub
            activeclass="primary--text"
          >
            <vlisttiletitle-stub>
              Sign out
            </vlisttiletitle-stub>
          </vlisttile-stub>
        </vlist-stub>
      </vmenu-stub>
    </vlisttileaction-stub>
  </vlisttile-stub>
   
  <vdivider-stub />
</vlist-stub>
`;
